chirp/drivers/icr6.py

```diff
@@ -0,0 +1,794 @@
+    _bank_class = ICR6Bank
+
+    @classmethod
+    def get_prompts(cls):
+        rp = chirp_common.RadioPrompts()
+        rp.experimental = ("This radio driver is currently under development, "
+                           "and not all the features or functions may work as"
+                           "expected. You should proceed with caution.")
+        return rp
+
+    def get_features(self):
+        rf = chirp_common.RadioFeatures()
+        rf.memory_bounds = (0, 1299)
+        rf.valid_modes = list(MODES)
+        rf.valid_tmodes = list(TONE_MODES)
+        rf.valid_duplexes = list(DUPLEX_DIRS)
+        rf.valid_bands = [(100000, 1309995000)]
+        rf.valid_skips = ["", "S", "P"]  # Not 1:1 mapping
+        rf.valid_characters = ICR6_CHARSET
+        rf.valid_name_length = NAME_LENGTH
+        rf.can_delete = True
+        rf.has_ctone = True
+        rf.has_dtcs = False           # TODO
+        rf.has_dtcs_polarity = False  # TODO
+        rf.has_bank = False           # TODO
+        rf.has_bank_names = False     # TODO
+        rf.has_name = True
+        rf.has_settings = True
+        rf.has_tuning_step = False  # hide in GUI, manage by code
+        # rf.valid_tuning_steps = list(STEPS)
+        # Attenuator = True
+        return rf
+
+    def process_mmap(self):
+        self._memobj = bitwise.parse(mem_format, self._mmap)
+
+    def get_raw_memory(self, number):
+        return repr(self._memobj.memory[number])
+
+    def get_memory(self, number):
+        _mem = self._memobj.memory[number]
+        _flag = self._memobj.flags[number]
+
+        mem = chirp_common.Memory()
+        mem.number = number
+
+        if _flag.hide_channel == 1:
+            mem.empty = True
+            return mem
+        mem.empty = False
+
+        if _mem.freq_flags == 0:
+            mem.freq = 5000 * (_mem.freq2 * 256 * 256 +
+                               _mem.freq1 * 256 +
+                               int(_mem.freq0))
+            mem.offset = 5000 * (_mem.offset_h * 256 + _mem.offset_l)
+        elif _mem.freq_flags == 20:
+            mem.freq = 6250 * (_mem.freq2 * 256 * 256 +
+                               _mem.freq1 * 256 +
+                               int(_mem.freq0))
+            mem.offset = 6250 * ((_mem.offset_h * 256) + _mem.offset_l)
+        elif _mem.freq_flags == 40:
+            mem.freq = 8333.3333 * (_mem.freq2 * 256 * 256 +
+                                    _mem.freq1 * 256 +
+                                    int(_mem.freq0))
+            mem.offset = 8333.3333 * ((_mem.offset_h * 256) + _mem.offset_l)
+        elif _mem.freq_flags == 60:
+            mem.freq = 9000 * (_mem.freq2 * 256 * 256 +
+                               _mem.freq1 * 256 +
+                               int(_mem.freq0))
+            mem.offset = 9000 * ((_mem.offset_h * 256) + _mem.offset_l)
+        else:
+            print(f"Unknown freq multiplier: {_mem.freq_flags}")
+            mem.freq = 1234567890
+            mem.offset = 0
```

Comment:
It would be better to raise an exception here. If anyone ever hits this, they're just going to file a bug saying "I put in 146.52 and the radio shows 123.45". Raise an exception explaining the problem so they'll see it.

Reply:
Fixed